python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1303 @@
+
+        This method must be defined as the dataset does not exist before this task is run.
+
+        Parameters
+        ----------
+        namespace
+            Results of parsing command-line (with ``butler`` and ``log`` elements).
+
+        Notes
+        -----
+        Not called if ``add_id_argument`` called with ``doMakeDataRefList=False``.
+        Note that this is almost a copy-and-paste of the vanilla implementation, but without checking
+        if the datasets already exist as this task exists to make them.
+        """
+        if self.datasetType is None:
+            raise RuntimeError("Must call setDatasetType first")
+        butler = namespace.butler
+        for dataId in self.idList:
+            refList = list(butler.subset(datasetType=self.datasetType, level=self.level, dataId=dataId))
+            # exclude nonexistent data
+            # this is a recursive test, e.g. for the sake of "raw" data
+            if not refList:
+                namespace.log.warn("No data found for dataId=%s", dataId)
+                continue
+            self.refList += refList
+
+
+class MakeBrighterFatterKernelTask(pipeBase.CmdLineTask):
+    """Brighter-fatter effect correction-kernel calculation task.
+
+    A command line task for calculating the brighter-fatter correction
+    kernel from pairs of flat-field images (with the same exposure length).
+
+    The following operations are performed:
+
+    - The configurable isr task is called, which unpersists and assembles the
+      raw images, and performs the selected instrument signature removal tasks.
+      The appropriacy of the selected isr components is checked.
+
+    - The gain of the each amplifier in the detector is calculated using
+      the PTC method and used to correct the images so that all calculations
+      are done in units of electrons, and so that the level across amplifier
+      boundaries is continuous. Outliers in the PTC are iteratively rejected
+      before fitting, with the nSigma rejection level set by
+      config.nSigmaClipRegression. Individual pixels are masked from
+      the image based on config.nSigmaClipGainCalc
+
+    - Each image is then cross-correlated with the one it's paired with
+      (with the pairing defined by the --visitPairs command line argument),
+      which is done either the whole-image to whole-image,
+      or amplifier-by-amplifier, depending on the config.level.
+
+    - Once the cross-correlations have been calculated for each visit pair,
+      these are used to generate the correction kernel.
+      Outlier values in these cross-correlations are rejected by using a
+      pixel-wise sigma-clipped thresholding to each cross-correlation in
+      the visitPair-length stack or cross-correlations.
+      The number of sigma clipped to is set by config.nSigmaClipKernelGen.
+      The maximum lag used, in pixels, and hence the size of the half-size
+      of the kernel generated, is given by config.maxLag,
```

Comment:
I suggest removing "in pixels" here, since it's documented in the config

Reply:
I dunno... I think that it reads as it should, and might save people searching around if they're confused. I think it adds clarity, sounds right, and as it will never change having it defined in multiple places is no problem. Unless you feel strongly that it should go I'd rather leave that there. However, as this is defining the shape of the output of the whole task I've moved it up in this section, and I think the whole thing reads a lot better now. Fro convenience, this section now reads:
```
    - Once the cross-correlations have been calculated for each visit pair,
      these are used to generate the correction kernel.
      The maximum lag used, in pixels, and hence the size of the half-size
      of the kernel generated, is given by config.maxLag,
      i.e. a value of 10 will result in a kernel of size 2n-1 = 19x19 pixels.
      Outlier values in these cross-correlations are rejected by using a
      pixel-wise sigma-clipped thresholding to each cross-correlation in
      the visitPair-length stack of cross-correlations.
      The number of sigma clipped to is set by config.nSigmaClipKernelGen.
```
Let me know if that's OK, or if you want changes (or the mention of "in pixels" removed).